src/wp-includes/rest-api/endpoints/class-wp-rest-block-patterns-controller.php

```diff
@@ -0,0 +1,198 @@
+
+	/**
+	 * Constructs the controller.
+	 *
+	 * @since 6.0.0
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'block-patterns/patterns';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since 6.0.0
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks whether a given request has permission to read block patterns.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) {
+		if ( current_user_can( 'edit_posts' ) ) {
+			return true;
+		}
+
+		foreach ( get_post_types( array( 'show_in_rest' => true ), 'objects' ) as $post_type ) {
+			if ( current_user_can( $post_type->cap->edit_posts ) ) {
+				return true;
+			}
+		}
+
+		return new WP_Error(
+			'rest_cannot_view',
+			__( 'Sorry, you are not allowed to view the registered block patterns.' ),
+			array( 'status' => rest_authorization_required_code() )
+		);
+	}
+
+	/**
+	 * Retrieves all block patterns.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		// Load block patterns from w.org.
+		_load_remote_block_patterns(); // Patterns with the `core` keyword.
+		_load_remote_featured_patterns(); // Patterns in the `featured` category.
+		_register_remote_theme_patterns(); // Patterns requested by current theme.
+
+		$response = array();
+		$patterns = WP_Block_Patterns_Registry::get_instance()->get_all_registered();
+		foreach ( $patterns as $pattern ) {
+			$prepared_pattern = $this->prepare_item_for_response( $pattern, $request );
+			$response[]       = $this->prepare_response_for_collection( $prepared_pattern );
+		}
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Prepare a raw block pattern before it gets output in a REST API response.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param object          $item    Raw pattern as registered, before any changes.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$fields = $this->get_fields_for_response( $request );
+		$keys   = array(
+			'name',
+			'title',
+			'description',
+			'viewportWidth',
```

Comment:
I guess it is is too late now, but the pattern we've been following is for the REST API to use `snake_case`. See the Block Types controller for instance.

Reply:
I think it should be possible, and implemented it now in this Gutenberg PR: https://github.com/WordPress/gutenberg/pull/40254

The patch changes the REST endpoint to respond with `block_types` and `viewport_width` field, and also changes the `core` data store to convert them back to `blockTypes` and `viewportWidth` when receiving the response. So that consumers of `wp.data.select( 'core' ).getBlockPatterns()` see the same `blockTypes` and `viewportWidth` fields as they did before.

@gziolo What would be the mechanics of including this in WP 6.0 beta? The REST endpoint part can be added as another commit in this PR, but the `wp.data` part needs to get there separately.

Reply:
We are going to cherry-pick https://github.com/WordPress/gutenberg/pull/40254 and https://github.com/WordPress/gutenberg/pull/40252 to `release/13.0` branch in Gutenberg and publish packages to npm in 1-2 hours.

Reply:
Addressed with https://core.trac.wordpress.org/changeset/53154 that was based on https://github.com/WordPress/wordpress-develop/pull/2574.